Assets/HoloToolkit/UX/Scripts/Lines/Bezeir.cs

```diff
@@ -0,0 +1,132 @@
+using UnityEngine;
+
+namespace MRTK.UX
+{
+    public class Bezeir : LineBase
+    {
+        [Serializable]
+        public struct PointSet
+        {
+            public PointSet (float spread)
+            {
+                Point1 = Vector3.right * spread * 0.5f;
+                Point2 = Vector3.right * spread * 0.25f;
+                Point3 = Vector3.left * spread * 0.25f;
+                Point4 = Vector3.left * spread * 0.5f;
+            }
+
+            public Vector3 Point1;
+            public Vector3 Point2;
+            public Vector3 Point3;
+            public Vector3 Point4;
+        }
+
+        [Header("Bezeir Settings")]
+        public PointSet Points = new PointSet(0.5f);
+
+        public override int NumPoints
+        {
+            get
+            {
+                return 4;
+            }
+        }
+
+        protected override Vector3 GetPointInternal(int pointIndex)
+        {
+            switch (pointIndex)
+            {
+                case 0:
+                    return Points.Point1;
+
+                case 1:
+                    return Points.Point2;
+
+                case 2:
+                    return Points.Point3;
+
+                case 3:
+                    return Points.Point4;
+
+                default:
+                    return Vector3.zero;
+            }
+        }
+
+        protected override void SetPointInternal(int pointIndex, Vector3 point)
+        {
+            switch (pointIndex)
+            {
+                case 0:
+                    Points.Point1 = point;
+                    break;
+
+                case 1:
+                    Points.Point2 = point;
+                    break;
+
+                case 2:
+                    Points.Point3 = point;
+                    break;
+
+                case 3:
+                    Points.Point4 = point;
+                    break;
+
+                default:
+                    break;
+            }
+        }
+
+        protected override Vector3 GetPointInternal(float normalizedDistance)
+        {
+            return LineUtils.InterpolateBezeirPoints(Points.Point1, Points.Point2, Points.Point3, Points.Point4, normalizedDistance);
+        }
+
+        protected override float GetUnclampedWorldLengthInternal()
+        {
+            // Crude approximation
+            // TODO optimize
+            float distance = 0f;
+            Vector3 last = GetUnclampedPoint(0f);
+            for (int i = 1; i < 10; i++)
+            {
+                Vector3 current = GetUnclampedPoint((float)i / 10);
+                distance += Vector3.Distance(last, current);
+            }
+            return distance;
+        }
+
+        protected override Vector3 GetUpVectorInternal(float normalizedLength)
+        {
+            // Bezeir up vectors just use transform up
+            return transform.up;
+        }
+
+#if UNITY_EDITOR
+        [UnityEditor.CustomEditor(typeof(Bezeir))]
+        public class CustomEditor : LineBaseEditor
```

Comment:
Nit: Editor scripts should be in their own classes.

Reply:
I think this one's worth discussing. There are benefits to keeping some editors nested under the scripts they target like this.

Eg the latest [Spline](https://github.com/Railboy/MixedRealityToolkit-Unity/blob/9b290c185bbcef999e68e585fb1a1da104f594d5/Assets/HoloToolkit/UX/Scripts/Lines/Spline.cs) class, where 'points' is now private. Other scripts are forced to access these points via GetPoint and SetPoint, which keeps things clean. But the Editor script still has direct access to the 'points' array, allowing it to do some error-checking stuff that wouldn't be possible otherwise.

To get the same functionality with a separated Editor you'd have to define editor-only public properties or some other potentially confusing kludge.

Devs also benefit from seeing that a custom editor has been declared and from seeing the logic inside it without having to go on a scavenger hunt.